dart/gui/CMakeLists.txt

```diff
@@ -62,6 +78,25 @@ add_subdirectory(osg)
 
 # Generate header for this namespace
 dart_get_filename_components(header_names "gui headers" ${hdrs})
+
+# Remove deprecated files from the list
+list(REMOVE_ITEM header_names
```

Comment:
> I think it's inadvertently breaking the API.

This is a valid point.

> We might want to consider allowing these to remain in the master header, even if it shoots compiler messages at people who are using the master `gui` header.

I have a concern that the user will always see the warning messages by just including the master `gui` header. I can think of two possible solutions to this:

1. Remove header warnings completely
Easiest solution. The deprecated header will be removed in the next major release. The downside of this method is obviously that the user won't get a prior notice the header will be removed even when the header is included explicitly (not by the master header).

2. Suppress header warning when it's included by a master header
This would require some work for us. We could use a processor of whether the deprecated header is included by the master header, and suppress the warning if so. One caveat of this is that it's possible to unintentionally suppress the warning if a deprecated header is firstly included by the master header and then included explicitly. This is because the second header inclusion would be just ignored by the header guard. One workaround would be putting the warning out side of the header guard, but this would pollute the output window too much.

Maybe there is a much cleaner way, which I cannot think of now. Let me know.